test/Twilio.Test/Security/RequestValidatorTest.cs

```diff
@@ -7,24 +8,54 @@ namespace Twilio.Tests.Security
     [TestFixture]
     public class RequestValidatorTest
     {
+        private readonly string url = "https://mycompany.com/myapp.php?foo=1&bar=2";
+        private readonly string body = "{\"property\": \"value\", \"boolean\": true}";
+        private readonly string bodySignature = "Ch/3Y02as7ldtcmi3+lBbkFQKyg6gMfPGWMmMvluZiA=";
+        private NameValueCollection parameters = new NameValueCollection();
+        private RequestValidator validator = new RequestValidator("12345");
 
-        [Test]
-        public void TestValidate()
+        public RequestValidatorTest()
         {
-            var validator = new RequestValidator("12345");
+            // Intentionally out of alphabetical order
+            parameters.Add("CallSid", "CA1234567890ABCDE");
+            parameters.Add("From", "+14158675309");
+            parameters.Add("Digits", "1234");
+            parameters.Add("To", "+18005551212");
+            parameters.Add("Caller", "+14158675309");
+        }
 
-            const string url = "https://mycompany.com/myapp.php?foo=1&bar=2";
-            var parameters = new Dictionary<string, string>
+        [Test]
+        public void TestValidateDictionary()
+        {
+            var dict = new Dictionary<string, string>();
+            foreach (var k in parameters.AllKeys)
             {
-                {"CallSid", "CA1234567890ABCDE"},
-                {"Caller", "+14158675309"},
-                {"Digits", "1234"},
-                {"From", "+14158675309"},
-                {"To", "+18005551212"}
-            };
+                dict.Add(k, parameters[k]);
+            }
+
+            const string signature = "RSOYDt4T1cUTdK1PDd93/VVr8B8=";
+            Assert.IsTrue(validator.Validate(url, parameters, signature), "Request does not match provided signature");
+        }
 
+        [Test]
+        public void TestValidateCollection()
+        {
             const string signature = "RSOYDt4T1cUTdK1PDd93/VVr8B8=";
             Assert.IsTrue(validator.Validate(url, parameters, signature), "Request does not match provided signature");
         }
+
+        [Test]
+        public void TestValidateBody()
+        {
+            Assert.IsTrue(validator.ValidateBody(body, bodySignature));
+        }
+
+        [Test]
+        public void TestValidateWithBody()
+        {
+            parameters.Add("bodySHA256", bodySignature);
```

Comment:
maybe add a test case without the required param? (I think it will return `false`, which is what I'd expect, but it seems like it's worth having it.)

Reply:
Done! Also added another negative test case.